### ISSUE TITLE
What about `require-button-type` with associated `role`?

### ISSUE THREAD

**Issue by gossi** [ember-template-lint/ember-template-lint#2349]
I have this HTML snippet for a tab-control:

```html
<button role="tab" ...>Label</button>
```

there is no `type="button"` attribute present, which causes the `require-button-type` rule to report this.

I am wondering if my understanding is wrong or if this a valid bug report? Afaik the `role` gives the advanced meaning and is "overwriting" the button behavior from whatever you would set as `type`, as a result, the `type` can be omitted, because a `role` is present. I'm following these guides as per my implementation:

- https://www.w3.org/TR/wai-aria-practices/#tabpanel
- https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-1/tabs.html
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role

I couldn't find a spec that defines the way in which order to process information for assisstive technology.

We are on `ember-template-lint`: `v3.10.0`

**Comment by bmish** [comment]
CC: rule author @Turbo87 (original discussion in #867).

**Comment by Turbo87** [comment]
what happens if you do this:

```html
<form action="foo">
  <button role="tab">bar</button>
</form>
```

if it submits the form then you still need to set `type="button"`, otherwise you might be correct :)

**Comment by gossi** [comment]
I checked @Turbo87 - it will submit the form. And as this might be in a component that potentially is used inside a form, where it accidentally/unexpectedly might submit the form, therefore it must have `type="button"` - got it.

Thank you, case closed.
